terraform/README.md

```diff
@@ -0,0 +1,154 @@
+# Terraform
+
+The persistent AWS infrastructure is deployed as Cloud.gov assets via Terraform. For a description of persistent infrastructure and how it differs from ephemeral infrastructure see the root README.md.
+
+These docs are verbose because this is technology with which developers will rarely interact. I suggest you settle in for a nice long read with your favorite drink of choice.
+
+## Set up
+
+1. **Install terraform**
+
+    - On MacOS: `brew install terraform`
+    - On other platforms: [Download and install terraform][tf]
+
+1. **Use githook for formatting**
+
+    Terraform has a specific whitespace formatting style that is difficult to maintain. Terraform includes a formatting command, `terraform fmt`, to help developers maintain the correct style. This repository contains a pre-commit hook that runs `terraform fmt` on all staged files so you don't have to remember to run this command.
+
+    If you are not using your own custom pre-commit hooks:
+
+    ```bash
+    # start from repo root directory
+
+    # make the pre-commit file executable
+    chmod 755 .githooks/pre-commit
+
+    # change your default hooks directory to `.githooks`.
+    git config core.hooksPath .githooks
+    ```
+
+    If you are already using git hooks, add the `.githooks/pre-commit` contents to your hooks directory or current pre-commit hook. Remember to make the file executable.
+
+1. **Install Cloud Foundry CLI tool**
+
+    - On MacOS: `brew install cloudfoundry/tap/cf-cli`
+    - On other platforms: [Download and install cf][cf-install]
+
+1. **Add target environment credentials**
+
+    We are using Terraform to create Cloud Foundry resources in a Cloud.gov account. Creating infrastructure on the Cloud.gov platform requires a Cloud.gov service account username and password. These keys are specific to each cloud.gov organization/space, i.e. each deployment environment, and can be generated by developers who have proper Cloud.gov permissions at any time.
+
+    You can use the same account credentials that are used to deploy applications via CircleCi or follow the steps below to generate a new set of credentials. If you need more information, check out the [service account docs][cloudgov-deployer].
```

Comment:
> You can use the same account credentials that are used to deploy applications via CircleCi

Do we want people to use the deploy credentials or just instruct everyone to generate their own credentials?

Reply:
It's probably cleaner if I just tell people to make their own separate account credentials.  That way if the deployer credentials need to be rotated for some reason people won't have their tf stop working. I'll update this.

Reply:
updated!